hack/local-development/start-admission-controller

```diff
@@ -16,9 +16,13 @@
 
 source $(dirname "${0}")/common/helpers
 
-mktemp_kubeconfig
-kubeconfig="$__tmp_kubeconfig"
-trap cleanup_kubeconfig EXIT
+if [[ $(k8s_env) == "$NODELESS" ]]; then
```

Comment:
Why is this change only done for `$NODELESS` and not for the other environment types, e.g. `$REMOTE`? Wouldn't it have some consequences on testing locally using these environment types, e.g. perhaps seed authorizer not working?

Reply:
> Why is this change only done for $NODELESS and not for the other environment types, e.g. $REMOTE?

Mainly due to https://github.com/gardener/gardener/pull/3901#discussion_r618116532.

> Wouldn't it have some consequences on testing locally using these environment types, e.g. perhaps seed authorizer not working?

WDYM? Seed Authorizer does only work in local-garden because only here we can configure the kube-apiserver with the auth-webhook flags.

Reply:
A `remote-garden` is usually just a shoot in our `dev` landscape, or in the general case a cluster that the developer owns. In this case, why wouldn't it be possible to configure `kube-apiserver` with the right flags? OK, it's probably not something we support right now for shoots, but is it just functionality that is missing or a more general limitation?

Reply:
As discussed via phone, it's probably rather a general limitation (of managed Kubernetes services in general), i.e., we would not want to enable end-users to inject auth webhooks, especially because this would mean that we needed to run untrusted code in the seed (next to the shoot's control plane) for low latency. There might be many other drawbacks and limitations, so we don't plan to invest into it which makes the remote garden development as outlined above quite complicated (assuming that such remote clusters are shoots).

Reply:
We discussed this. Even though it could be possible to add such functionality for shoots we will probably never do it due to the security and other implications. So even if it's enabled for `$REMOTE`, there would be no easy way for us to take advantage of it.

We also thought it would be a good idea to add a sentence to the documentation stating that the remote garden setup doesn't support the seed authorizer and therefore should be used with care.